### PR TITLE
Python 3: struct.pack() update

### PR DESCRIPTION
**Pull request by hereischen**
Signed-off-by: Haotong Chen <hachen@redhat.com>

### CODE REVIEW

File: shared/deps/serial/VirtIoChannel_guest_send_receive.py [hunk 1]
```diff
@@ -104,7 +105,9 @@ def shake_hand(self, size=0, action="receive"):
             print("xxxx size = %s" % size)
             if size:
                 size = struct.unpack("q", size)[0]
-                txt = struct.pack(self.ack_format, self.ack_msg)
+                if isinstance(self.ack_msg, six.string_types):
+                    ack_msg_bytes = self.ack_msg.encode()
+                txt = struct.pack(self.ack_format, ack_msg_bytes)
```
Comment (luckyh) [inline shared/deps/serial/VirtIoChannel_guest_send_receive.py]: Should be:
```python
ack_msg_bytes = self.ack_msg
if isinstance(self.ack_msg, six.string_types):
    ack_msg_bytes = ack_msg_bytes.encode()
txt = struct.pack(self.ack_format, ack_msg_bytes)
```

File: shared/deps/run_autotest/boottool.py [hunk 2]
```diff
@@ -199,6 +199,9 @@ def get_packed(self):
         params.append(0)
         params.append(self.attributes)
 
+        if isinstance(params[0], six.string_types):
+            params[0] = params[0].encode()
```
Comment (luckyh) [inline shared/deps/run_autotest/boottool.py]: All the string elements of `params` should be encoded.

https://docs.python.org/3.6/library/struct.html?highlight=struct%20pack#struct.pack

File: virttest/ppm_utils.py [hunk 3]
```diff
@@ -250,7 +250,8 @@ def image_comparison(width, height, data1, data2):
         else:
             # Not equal -- give the pixel a reddish hue
             newpixel = [value, 0, 0]
-        newdata += struct.pack("BBB", newpixel[0], newpixel[1], newpixel[2])
+        newdata += struct.pack("BBB", newpixel[0].encode(),
+                               newpixel[1], newpixel[2])
```
Comment (luckyh) [inline virttest/ppm_utils.py]: The same as the above.

File: virttest/utils_net.py [hunk 4]
```diff
@@ -441,7 +441,8 @@ def get_mac(self):
         '''
         Obtain the device's mac address.
         '''
-        ifreq = struct.pack('16sH14s', self.name, socket.AF_UNIX, '\x00' * 14)
+        ifreq = struct.pack('16sH14s', self.name.encode(),
+                            socket.AF_UNIX, '\x00' * 14)
```
Comment (luckyh) [inline virttest/utils_net.py]: Should be `b\x00`, right?

File: shared/deps/serial/VirtIoChannel_guest_send_receive.py [hunk 5]
```diff
@@ -12,6 +12,7 @@
 import struct
 import platform
 import optparse
+import six
```
Comment (luckyh) [inline shared/deps/serial/VirtIoChannel_guest_send_receive.py]: Guest may not have `six` installed.

File: virttest/ppm_utils.py [hunk 6]
```diff
@@ -250,7 +250,8 @@ def image_comparison(width, height, data1, data2):
         else:
             # Not equal -- give the pixel a reddish hue
             newpixel = [value, 0, 0]
-        newdata += struct.pack("BBB", newpixel[0], newpixel[1], newpixel[2])
+        newdata += struct.pack("BBB", newpixel[0].encode(),
+                               newpixel[1].encode(), newpixel[2].encode())
```
Comment (luckyh) [inline virttest/ppm_utils.py]: Checked the context, all the elements of the list `newpixel` are `int`.

File: virttest/rss_client.py [hunk 7]
```diff
@@ -233,6 +233,8 @@ def _receive_file_chunks(self, filename, timeout=60):
             f.close()
 
     def _send_msg(self, msg, timeout=60):
+        if isinstance(msg, six.string_types):
+            msg = msg.encode()
```
Comment (luckyh) [inline virttest/rss_client.py]: Checked the context, the `msg` should be `int`, so not need to change this.

File: shared/deps/run_autotest/boottool.py [hunk 8]
```diff
@@ -199,6 +199,10 @@ def get_packed(self):
         params.append(0)
         params.append(self.attributes)
 
+        for idx, item in enumerate(params):
+            if isinstance(item, six.string_types):
+                params[idx] = item.encode()
```
Comment (luckyh) [inline shared/deps/run_autotest/boottool.py]: Seems that the list has no `string`/`bytes` element inside.

File: virttest/utils_net.py [hunk 9]
```diff
@@ -603,7 +607,8 @@ def dellink(self):
         sock.send(self.__netlink_pack(msgtype=arch.RTM_DELLINK,
                                       flags=arch.NLM_F_REQUEST | arch.NLM_F_ACK,
                                       seq=1, pid=0,
-                                      data=struct.pack('BxHiII', arch.AF_PACKET,
+                                      data=struct.pack('BxHiII',
+                                                       arch.AF_PACKET.encode(),
```
Comment (luckyh) [inline virttest/utils_net.py]: `arch.AF_PACKET` is `int`.

File: shared/deps/serial/VirtIoChannel_guest_send_receive.py (hunk 5 above)
Comment (hereischen) [reply shared/deps/serial/VirtIoChannel_guest_send_receive.py]: updated, thanks

**Comment by chunfuwen** [comment]
@hereischen 
@luckyh 
utils_net.py is commonly used frequently, do we have opportunities to have basic validation after changes?

**Comment by hereischen** [comment]
@chunfuwen I have tried it on both python 2 and 3 running acceptance.
Do you have a issues related to this change?

**Pull request merged**